### PR TITLE
Removed circular reference from website security

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

~~This PR moves the dependency to the `RequestAnalyzer` from the `UserRepository` to the `AuthenticationHandler`. That means that the check on the `Sulu` system is fixed in the `AuthenticationHandler`.~~ We have decided to remove the `RequestAnalyzer` dependency here, and use the data directly written on the request instead.

#### Why?

This resolves the circular reference issue we were having on the website (if it was secured).

#### BC Breaks/Deprecations

The system specific methods from the `UserRepository` were removed, and the `UserRepositoryFactory` (on which hopefully nobody relied) was removed.

#### To Do

- [x] Add `UPGRADE.md``
- [x] ~~Add `WebspaceAuthenticationHandler` for usage with the website (?)~~

### CODE REVIEW

**Comment by danrot** [comment]
@chirimoya Should I also add an `AuthenticationHandler` to be used on the website? This one would only be activated if needed, and would only act as a default. It would use the system value from the webspace in combination with the `RequestAnalyzer`. However, I am not sure if it is worth it, since it would not be used often, but still be located in the core library, and some project might want to write their own `AuthenticationHandler`s anyway. But it would definitely be a valuable default (and if we don't add it, the security system in the webspace xml wouldn't make too much sense).

TL;DR: I would slightly prefer to add this.

And btw, do you like this solution at all? 😁 Didn't find a better way to get rid of the `RequestAnalyzer` dependency otherwise (splitting the `UserRepository` was not enough). /cc @wachterjohannes

File: src/Sulu/Bundle/SecurityBundle/Resources/config/services.xml [hunk 1]
```diff
@@ -146,7 +140,9 @@
         </service>
 
         <service id="sulu_security.user_repository" class="%sulu_security.user_repository.class%">
-            <factory service="sulu_security.user_repository_factory" method="getRepository"/>
+            <factory service="doctrine.orm.entity_manager" method="getRepository"/>
+
+            <argument>SuluSecurityBundle:User</argument>
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Resources/config/services.xml]: use the container parameter for that. otherwise it wouldn't work for overridden users.

File: src/Sulu/Bundle/SecurityBundle/Entity/UserRepository.php [hunk 2]
```diff
@@ -410,29 +257,6 @@ public function getUsernameByApiKey($apiKey)
     }
 
     /**
-     * returns all users within the defined system including their contacts.
-     */
-    public function getUserInSystem()
```
Comment (chirimoya) [inline src/Sulu/Bundle/SecurityBundle/Entity/UserRepository.php]: Check with @christianbader if really not needed any more?

File: src/Sulu/Bundle/SecurityBundle/User/UserProvider.php [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Exception\DisabledException;
+use Symfony\Component\Security\Core\Exception\LockedException;
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+/**
+ * Responsible for loading the user from the database for the Symfony security system. Takes also the configuration
```
Comment (chirimoya) [inline src/Sulu/Bundle/SecurityBundle/User/UserProvider.php]: Takes also the security system configuration ...

File: src/Sulu/Bundle/SecurityBundle/User/UserProvider.php [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        $this->requestStack = $requestStack;
+        $this->suluSystem = $suluSystem;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loadUserByUsername($username)
+    {
+        $message = sprintf(
```
Comment (chirimoya) [inline src/Sulu/Bundle/SecurityBundle/User/UserProvider.php]: `$exceptionMessage`
